infiniband-diags/ibstat.c

```diff
@@ -310,10 +312,14 @@ int main(int argc, char *argv[])
 		IBPANIC("can't list IB device names");
 
 	if (argc) {
-		for (node = device_list; node; node = node->next)
-			if (!strncmp(node->ca_name, argv[0],
```

Comment:
strcmp() instead of strncmp() will do the trick.

 15 int main()                                                                                                         
 16 {                                                                                                                 
 17         char str1[] = "abc";                                                                                      
 18         char str2[] = "abcd";                                                                                     
 19                                                                                                                   
 20         printf("%d\n", strcmp(str1, str2));                                                                       
 21         printf("%d\n", strcmp(str2, str1));                                                                       
 22                                                                                                                   
 23         return 0;                                                                                                 
 24 }

And add fixes line please.